### PR TITLE
docs: describe platform section in the config file

### PR DESCRIPTION
**Pull request by athos-ribeiro**

### CODE REVIEW

File: docs/configuration_file.md [hunk 1]
```diff
@@ -171,6 +173,12 @@ Some options are also mandatory.
 
 * `build_from` (*optional*, `string`) — build source to use, consists of 2 parts separated with delimiter ':', first part can be : image or imagestream, and second part is corresponding image or imagestream
 
+### `[platform:ARCH]` options
+
+* `architecture` (*optional*, `string`) — platform's GOARCH (Go language platform name). If not declared, this option assumes the value of `ARCH`.
```
Comment (twaugh) [inline docs/configuration_file.md]: This reads a little bit as though the default is the literal string "ARCH". Can we re-word it to say the default is the same as the platform name being defined?

File: docs/configuration_file.md [hunk 2]
```diff
@@ -171,6 +173,12 @@ Some options are also mandatory.
...
+
+* `enable_v1` (*optional*, `boolean`) — enable support for Docker Registy HTTP API v1 for a given platform. Only one platform should have this option enabled. Defaults to false.
```
Comment (twaugh) [inline docs/configuration_file.md]: Typo: Registy -> Registry

File: docs/configuration_file.md [hunk 3]
```diff
@@ -14,6 +14,8 @@ key=value
 
 There is section `[general]` which contains global configuration for all instances. You can put more sections to your config which represents multiple instances of OSBS (and then you can refer to them with command line option `--instance`).
 
+It is also possible to describe platforms in platform sections, which have the form `[platform:ARCH]`, where `ARCH` is the platform name, such as `x86_64` or `aarch64`. Platform sections are not interpreted as OSBS instances, but descriptions to be applied to OSBS instances running on the given platform.
```
Comment (twaugh) [inline docs/configuration_file.md]: Minor: could remove commas before "where" and "such as".

File: docs/configuration_file.md (hunk 2 above)
Comment (athos-ribeiro) [reply docs/configuration_file.md]: Thanks, I keep forgetting to run aspell on documentation changes. Got another typo on the run and included the fix in this patch.

File: docs/configuration_file.md (hunk 3 above)
Comment (athos-ribeiro) [reply docs/configuration_file.md]: done

File: docs/configuration_file.md (hunk 1 above)
Comment (athos-ribeiro) [reply docs/configuration_file.md]: done

**Comment by athos-ribeiro** [comment]
Documentation change only. No release notes entries added.

**Pull request merged**